engine/crates/runtime-noop/src/hooks.rs

```diff
@@ -1,19 +1,34 @@
-use std::{collections::HashMap, sync::Arc};
-
-use runtime::hooks::{HeaderMap, HookError, HooksImpl, UserError};
+use runtime::hooks::{HeaderMap, Hooks};
 
 #[derive(Clone)]
 pub struct HooksNoop;
 
-#[async_trait::async_trait]
-impl HooksImpl for HooksNoop {
-    type Context = HashMap<String, String>;
+impl Hooks for HooksNoop {
+    type Context = ();
+
+    async fn on_gateway_request(
+        &self,
+        headers: HeaderMap,
+    ) -> Result<(Self::Context, HeaderMap), runtime::error::GraphqlError> {
+        Ok(((), headers))
+    }
 
-    async fn on_gateway_request(&self, headers: HeaderMap) -> Result<(Self::Context, HeaderMap), HookError> {
-        Ok((HashMap::new(), headers))
+    async fn authorize_edge_pre_execution<'a>(
+        &self,
+        _: &Self::Context,
+        _: runtime::hooks::EdgeDefinition<'a>,
+        _: impl serde::Serialize + serde::de::Deserializer<'a> + Send,
+        _: impl serde::Serialize + serde::de::Deserializer<'a> + Send,
+    ) -> Result<(), runtime::error::GraphqlError> {
+        unreachable!("@authorization directive not available outside of local context")
```

Comment:
Could this be called if somebody deploys a schema with `@authorization` to a managed runtime on Cloudflare and we crash to unreachable. So this would not really be unreachable, it can be reached very easily.

Reply:
yeah you're right, changing that quickly